config/src/main/java/org/springframework/security/config/authentication/AuthenticationManagerBeanDefinitionParser.java

```diff
@@ -86,11 +89,16 @@ public BeanDefinition parse(Element element, ParserContext pc) {
 		if ("false".equals(element.getAttribute(ATT_ERASE_CREDENTIALS))) {
 			providerManagerBldr.addPropertyValue("eraseCredentialsAfterAuthentication", false);
 		}
-		// Add the default event publisher
-		BeanDefinition publisher = new RootBeanDefinition(DefaultAuthenticationEventPublisher.class);
-		String pubId = pc.getReaderContext().generateBeanName(publisher);
-		pc.registerBeanComponent(new BeanComponentDefinition(publisher, pubId));
-		providerManagerBldr.addPropertyReference("authenticationEventPublisher", pubId);
+
+		if (!pc.getRegistry().containsBeanDefinition(AUTHENTICATION_EVENT_PUBLISHER_BEAN_NAME)) {
+			// Add the default event publisher to the context
+			GenericBeanDefinition publisher = new GenericBeanDefinition();
```

Comment:
This should remain a `RootBeanDefinition`. The only code that needs to change is the addition of the `if` statement and the bean name.

Reply:
I think I got it the correct way now